sunpy/visualization/animator/wcs.py

```diff
@@ -299,10 +303,27 @@ def plot_start_image_2d(self, ax):
 
         return im
 
+    def get_2d_plot_clip_bounds(self):
+        """
+        Update vmin/vmax when set to 'auto'.
+        """
+        if self.imshow_kwargs.get('vmax') == 'auto':
+            clip_interval = (1, 90)*u.percent
+            clip_percentages = clip_interval.to('%')
+            clip_limits = AsymmetricPercentileInterval(*clip_percentages.value)
+            vmin, vmax = clip_limits.get_limits(self.data_transposed)
+            return vmin, vmax
+
     def update_plot_2d(self, val, im, slider):
         """
         Update the image plot.
         """
         self.axes.reset_wcs(wcs=self.wcs, slices=self.slices_wcsaxes)
         im.set_array(self.data_transposed)
+
+        bounds = self.get_2d_plot_clip_bounds()
+        if bounds is not None:
+            vmin, vmax = bounds
```

Comment:
I don't like the fact we have to check if its none and then unpack the return after the fact. 

The issue is that I don't have a solution to this.

Reply:
What if ```get_2d_plot_clip_bounds``` returns a ```[None, None] ```, that way  
```python
vmin, vmax = self.get_2d_plot_clip_bounds()
``` 
would update the plot as ```None```

Reply:
What happens if you pass in None,None to the limit function?

Reply:
Is this better instead of having to check for None and unpack?

Reply:
> What happens if you pass in None,None to the limit function?

I think this is a better way to go about it

```python
def get_2d_plot_clip_bounds(self):
    """
    Update vmin/vmax when set to 'auto'.
    """
    vmin, vmax = None, None
    if self.imshow_kwargs.get('vmax') == 'auto':
        clip_interval = (1, 90)*u.percent
        clip_percentages = clip_interval.to('%')
        clip_limits = AsymmetricPercentileInterval(*clip_percentages.value)
        vmin, vmax = clip_limits.get_limits(self.data_transposed)
        return vmin, vmax

def update_plot_2d(self, val, im, slider):
    """
    Update the image plot.
    """
    self.axes.reset_wcs(wcs=self.wcs, slices=self.slices_wcsaxes)
    im.set_array(self.data_transposed)

    vmin, vmax = self.get_2d_plot_clip_bounds()
    im.set_clim(vmin, vmax)

    slider.cval = val
```

Reply:
Hmmm, I like it more than the current way. I think the name of the function still needs some work.

Reply:
I'm can't really thing of anything better. :sweat_smile:

Reply:
For the function name? Well it doesn't get the bounds of an image, and it only works if vmax is set to auto.  

It currently gives the impression it will update the bounds of an image every step.

Reply:
changed it to ```get_2d_plot_limts``` hoping that it is better :sweat_smile: